api/src/main/java/org/openmrs/module/pihcore/apploader/apps/patientregistration/SectionsSierraLeone.java

```diff
@@ -115,6 +115,23 @@ private Question getLocalContactPhoneNumber() {
 
         return q;
     }
+    /**
+     * In the base class this method excludes the Country field.
+     * In Sierra Leone we want to display the Country because we are capturing addresses from two other countries, Liberia and Guinea
+     * @param levels
+     * @return a String representing the full address
+     */
+    @Override
+    protected String getAddressHierarchyDisplayTemplate(List<AddressHierarchyLevel> levels) {
+        StringBuilder displayTemplate = new StringBuilder();
+        displayTemplate.append("{{nvl field.[" + levels.size() + "] '-'}}");
+        for (int i = levels.size() - 1; i >= 1; i--) {
+            displayTemplate.append(", {{field.[" + i + "]}}");
+        }
+
+        return displayTemplate.toString();
+    }
+
```

Comment:
This seems fine.

If we really want to try to improve on this overall and for the future, might opt instead to just modify the original method to support both use cases.  For example, this method could take in a boolean `showCountry` parameter, and if `true` then iterate down to level 1, and if `false` then iterate down to level 2.

Then, the `getLocalAddressQuestion` and `getAddressQuestion` methods could pull this `showCountry` configuration from the registration config or (even better) determine what to do based on whether more than one country are included in the address hierarchy configuration.

Reply:
Thanks @mseaton ! I have updated this PR with your suggestion. I also opened a SL config with the change:
https://github.com/PIH/openmrs-config-pihsl/pull/82